lib/sonata/staking/executeChannelStake.ts

```diff
@@ -0,0 +1,37 @@
+import getUser from "@/lib/neynar/getNeynarUser";
+import { stack } from "@/lib/stack/client";
+import supabase from "@/lib/supabase/serverClient";
+import getChannelTipInfo from "../tip/getChannelTipInfo";
+
+async function executeChannelStake(channelId: string, stakeAmount: number, fid: number) {
+  const user = await getUser(fid);
+  if (!user?.verifications?.[0]) throw Error('No user address found');
+
+  const balances: { amount: number; address: string }[] = await stack.getPoints(
+    user.verifications.map((verification: string) => verification)
+  );
+  const balance = balances.find(balance => balance.amount >= stakeAmount);
+  if (!balance) throw Error('No balances found');
+
+  const amount = Math.min(balance.amount, stakeAmount);
+
+  const info = await getChannelTipInfo(channelId, 0);
+  if (!info) throw Error('could not find channel');
+
+  const { channelAddress } = info;
+
+  const results = await Promise.all([
+    stack.track(`channel_stake_from_${fid}`, { account: balance.address, points: -amount }),
+    stack.track(`channel_stake_on_${channelAddress}`, { account: channelAddress, points: amount }),
```

Comment:
could we call this `to` instead of `on`?
- opposite of `from` is `to` 
```suggestion
    stack.track(`channel_stake_to_${channelAddress}`, { account: channelAddress, points: amount }),
```